### ISSUE TITLE
Flame chart output for rules

### ISSUE THREAD

**Issue by Mathnerd314** [ndmitchell/shake#480]
The [docs](https://github.com/ndmitchell/shake/blob/0f80d0199a9858da7e32af4f1437402f646e1113/docs/Profiling.md#chrome-traces) for Chrome traces currently say:
> Note that process 0 shows the time between a rule starting and finishing, and does not display time the rule was suspended waiting for its dependents.

This is a lie; in fact process 0 is just process 1 with the command names replaced with rule names; it does not use the rule timings.

Ideally, a Shake profile would contain all of start, stop, and duration, and show something like Firefox's [Flame chart](https://developer.mozilla.org/en-US/docs/Tools/Performance/Flame_Chart):

![flamechart](https://cloud.githubusercontent.com/assets/322214/16864212/dbb7669e-4a17-11e6-9701-8660cf91897d.png)

**Comment by ndmitchell** [comment]
You are right about the current information, and I've changed the docs. I am sure at one point it used to do that... But I clearly stopped it doing that at some point.

The flame chart looks like a very nice visualisation, and should be possible to recreate from the information stored. The only issue is that I don't immediately see how to scale it up to multithreaded builds, especially when children are demanded by two parents. I guess trying to recreate that as separate process numbers is probably the best approach?
